src/clj_icat_direct/queries.clj

```diff
@@ -294,6 +297,23 @@
   [uuid]
   (mk-paths-for-uuids [uuid]))
 
+(defn mk-perms-for-item
+  [path]
+  (let [[dirname basename] ((juxt #(.getParent %) #(.getName %)) (file path))]
+    (sql/format
+     (-> {:with [[:object-lookup {:union [(-> (h/select [:coll_id :object_id])
```

Comment:
doesn't need to be now, partly noting for myself but it'd apply to anyone who wants to do the refactor, it'd be nice to pull this object-lookup CTE code into a helper function and use it in at least `mk-get-item` (which needs more columns than this, but that could be parameterized). I'd like to refactor the listing stuff to use a strategy more like this too, so it'd apply there too eventually

(also, I like us using honeysql for this stuff. lots cleaner :+1: )

Reply:
I like the idea of splitting this into a separate function. I'll probably do that now. I agree about HoneySQL. It's a lot easier to deal with than long unwieldy string constants.